src/isa/mipsisainfo_common.h

```diff
@@ -0,0 +1,217 @@
+  SLTU = 0b101001,
+  JALR = 0b001001,
+  JR = 0b001000,
+  MFHI = 0b010000,
+  MFLO = 0b010010,
+  MTHI = 0b010001,
+  MTLO = 0b010011,
+  BREAK = 0b001101,
+  SYSCALL = 0b001100
+};
+
+} // namespace MIPSISA
+
+namespace MIPSABI {
+// MIPS ELF info
+// Elf flag masks
+enum MIPSElfFlags { NOREORDER = 0b1, PIC = 0b10, CPIC = 0b100, ARCH = 0b0 };
+extern const std::map<MIPSElfFlags, QString> ELFFlagStrings;
+
+enum SysCall {
+  None = 0,
+  PrintInt = 1,
+  PrintFloat = 2,
+  PrintDouble = 3,
+  PrintStr = 4,
+  ReadInt = 5,
+  ReadFloat = 6,
+  ReadDouble = 7,
+  ReadString = 8,
+  Sbrk = 9,
+  Exit = 10,
+  PrintChar = 11,
+  ReadChar = 12,
+  Open = 13,
+  Read = 14,
+  Write = 15,
+  Close = 16,
+  Exit2 = 17,
+  Time = 30,
+  MIDIout = 31,
+  Sleep = 32,
+  MIDIoutSynchronous = 33,
+  PrintIntHex = 34,
+  PrintIntBinary = 35,
+  PrintIntUnsigned = 36,
+  SetSeed = 40,
+  RandomInt = 41,
+  RandomIntRange = 42,
+  RandomFloat = 43,
+  RandomDouble = 44,
+  ConfirmDialog = 50,
+  InputDialogInt = 51,
+  InputDialogFloat = 52,
+  InputDialogDouble = 53,
+  InutDialogString = 54,
+  MessageDialog = 55,
+  MessageDialogInt = 56,
+  MessageDialogFloat = 57,
+  MessageDialogDouble = 58,
+  MessageDialogString = 59
+
+};
+
+} // namespace MIPSABI
+
+class MIPSISAInfoBase : public ISAInfoBase {
+public:
+  unsigned int regCnt() const override { return 34; }
+  QString regName(unsigned i) const override {
+    QString regNameDollar = MIPSISA::RegNames.size() > static_cast<int>(i)
+                                ? MIPSISA::RegNames.at(static_cast<int>(i))
+                                : QString();
+    regNameDollar.prepend("$");
+    regNameDollar.prepend("x");
+    return (MIPSISA::RegNames.size() > static_cast<int>(i)
+                ? MIPSISA::RegNames.at(static_cast<int>(i))
+                : QString());
+    // return regNameDollar;
+  }
+  QString regAlias(unsigned i) const override {
+    return MIPSISA::RegAliases.size() > static_cast<int>(i)
+               ? MIPSISA::RegAliases.at(static_cast<int>(i))
+               : QString();
+  }
+  QString regInfo(unsigned i) const override {
+    return MIPSISA::RegDescs.size() > static_cast<int>(i)
+               ? MIPSISA::RegDescs.at(static_cast<int>(i))
+               : QString();
+  }
+  QString name() const override { return CCmarch().toUpper(); }
+  bool regIsReadOnly(unsigned i) const override { return i == 0; }
+  int spReg() const override { return 29; }
+  int gpReg() const override { return 28; }
+  int syscallReg() const override { return 2; }
+  unsigned instrBits() const override { return 32; }
+  unsigned elfMachineId() const override { return EM_MIPS; }
+  unsigned int regNumber(const QString &reg, bool &success) const override {
+    QString regRes = reg;
+    QString regNoDollar = regRes.remove('$');
+
+    if (reg[0] != '$') {
+      success = false;
+      return 0;
+    }
+
+    success = true;
+    if (reg[0] == '$' && (MIPSISA::RegNames.count(reg) != 0)) {
```

Comment:
and compute reg without the dollar afterwords by removing the first character. 
Secondly, there is no need to include `reg[0] == '$'` here since you've already proved that in the prior if statement.

Reply:
```suggestion
    success = true;
    if (MIPSISA::RegNames.count(reg) != 0) {
```
Is this correct?